feature-proxy-impl/src/main/java/io/novafoundation/nova/feature_proxy_impl/data/common/RealProxyDepositCalculator.kt

```diff
@@ -0,0 +1,32 @@
+package io.novafoundation.nova.feature_proxy_impl.data.common
+
+import io.novafoundation.nova.common.utils.numberConstant
+import io.novafoundation.nova.common.utils.proxy
+import io.novafoundation.nova.feature_proxy_api.data.common.DepositBaseAndFactor
+import io.novafoundation.nova.feature_proxy_api.data.common.ProxyDepositCalculator
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.getRuntime
+import java.math.BigInteger
+
+class RealProxyDepositCalculator(
+    private val chainRegestry: ChainRegistry
+) : ProxyDepositCalculator {
+
+    override suspend fun getDepositConstants(chain: Chain): DepositBaseAndFactor {
+        val runtime = chainRegestry.getRuntime(chain.id)
+        val constantQuery = runtime.metadata.proxy()
+        return DepositBaseAndFactor(
+            baseAmount = constantQuery.numberConstant("ProxyDepositBase", runtime),
+            factorAmount = constantQuery.numberConstant("ProxyDepositFactor", runtime)
+        )
+    }
+
+    override fun calculateProxyDeposit(baseAndFactor: DepositBaseAndFactor, proxiesCount: Int): BigInteger {
```

Comment:
The name looks unclear to me - does this calculate the deposit for the account that ALREADY has `proxiesCount` or it calculates NEXT deposit given that account has `proxiesCount`?

Reply:
It seems to me that the semantics speak for themselves, I mean that this method doesn't take an account in arguments and it calculates the deposit for a certain number of proxies.
We can't know the next deposit since we don’t know whether this will be an operation of adding or removing a proxy